app/controllers/api/reports_controller.rb

```diff
@@ -40,6 +40,132 @@ def cat3
+      # table headers
+      sheet.row(6).push 'MRN', 'First Name', 'Last Name', 'Gender', 'Birthdate'
+      sheet.row(6).default_format = format
+      row = 7;
+      (0..4).each do |i| 
+        sheet.row(i).set_format(0, format) 
+      end
+      
+      records.each do |record|
+        value = record.value
+        authorize! :read, Record.find_by(medical_record_number: value[:medical_record_id])
+        if value["#{type}"] == 1
+          sheet.row(row).push value[:medical_record_id], value[:first], value[:last], value[:gender], Time.at(value[:birthdate]).strftime("%D")
+          row +=1
+        end
+      end
+
+      today = Time.now.strftime("%D")  
+      filename = "patients_" + measure.cms_id + "_" + patient_type(type) + "_" + "#{today}" + ".xls"
+      data = StringIO.new '';
+      book.write data;
+      send_data(data.string, {
+        :disposition => 'attachment',
+        :encoding => 'utf8',
+        :stream => false,
+        :type => 'application/excel',
+        :filename => filename
+      })
+    end
+
+    api :GET, '/reports/measures_spreadsheet', "Retrieve a spreadsheet of measure calculations"
+    param :username, String, :desc => 'Username of user to generate reports for'
+    param :effective_date, String, :desc => 'Time in seconds since the epoch for the end date of the reporting period'
+    param :provider_id, String, :desc => 'The Provider ID for spreadsheet generation', :required => true
+    description <<-CDESC
+      This action will generate an Excel spreadsheet document containing a list of measure calculations for the current user's selected measures.
+    CDESC
+    def measures_spreadsheet
+      book = Spreadsheet::Workbook.new
+      sheet = book.create_worksheet
+      format = Spreadsheet::Format.new :weight => :bold
+
+      user = User.where(:username => params[:username]).first || current_user
+      effective_date = params[:effective_date] || current_user.effective_date      
+      measure_ids = user.preferences['selected_measure_ids']
+      
+      unless measure_ids.empty?
+        selected_measures = measure_ids.map{ |id| HealthDataStandards::CQM::Measure.where(:id => id)}
+        # report header
+        provider = Provider.find(params[:provider_id])
+        authorize! :read, provider
+
+        eff = Time.at(params[:effective_date].to_i)
+        end_date = eff.strftime("%D")
+        start_date = eff.month.to_s + "/" + eff.day.to_s + "/" + (eff.year-1).to_s
+
+        sheet.row(0).push "Reporting Period: ", start_date + " - " + end_date
+        sheet.row(1).push "Provider: ", provider.full_name
+        sheet.row(2).push "NPI: ", provider.npi
+        # table headers
+        sheet.row(4).push 'NQF ID', 'CMS ID', 'Sub ID', 'Title', 'Subtitle', 'Numerator', 'Denominator', 'Exclusions', 'Percentage'
+        sheet.row(4).default_format = format
+        (0..2).each do |i|
+          sheet.row(i).set_format(0, format)
+        end
+        r = 5
```

Comment:
This is a magic number (http://en.wikipedia.org/wiki/Magic_number_%28programming%29). Maybe you could set r = 0 on line 130 and increment it as you go along or create a constant to signify that the end of the header is at 5.